core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/SearchModelImpl.java

```diff
@@ -245,6 +241,36 @@ private String getSearchConfigJson() {
 
     private Locale getLocale() {
         return Optional.ofNullable(resource.adaptTo(ResourceWrapper.class)).map(ResourceWrapper::getLocale)
-                .orElse(Locale.getDefault());
+                       .orElse(Locale.getDefault());
+    }
+
+    private ConnectionFailedAlert resolveConnectionFailedAlert() {
+        if (searchCAConfigurationModel == null) {
+            log.warn("Can not resolve context aware search configuration model.");
+            return new ConnectionFailedAlert(AlertVariant.WARNING,
+                                             Collections.singletonList(i18n.get(
+                                                     I18N_SEARCH_CA_CONFIGURATION_FAILED_TO_RESOLVE,
+                                                     null,
+                                                     resource.getPath())));
+        }
+        try {
+            boolean pingSuccess = fulltextSearchPingService.ping(searchCAConfigurationModel);
+            if (!pingSuccess) {
+                return new ConnectionFailedAlert(AlertVariant.ERROR,
+                                                 Arrays.asList(i18n.get(
+                                                                       I18N_SEARCH_CONNECTION_FAILED_FURTHER_ACTION_CHECK_OSGI_CONFIGURATION),
+                                                               i18n.get(
+                                                                       I18N_SEARCH_CONNECTION_FAILED_FURTHER_ACTION_CHECK_LOG_FILES)));
+            }
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
```

Comment:
Why do we wrap the exception?

Reply:
in order to display the message in the Connection Failed alert box

Reply:
aha, you are right, it is obsolete